### ISSUE TITLE
20-30s Support

### ISSUE THREAD

**Issue by PV-V** [BotoX/xiaomi-m365-compatible-bms#2]
Will this code work for 2x chip versions of BMS that support higher cell count (up to 30S)?
Thanks.

**Comment by BotoX** [comment]
Nope, not as it is right now anyways.
But it shouldn't be too hard to make that work, I don't have access to a BMS like that though.

**Comment by Hermit12** [comment]
Hi Botox,
I have a different issue but I dont want to open a new thread, its just a small question:
I made a 13S battery thanks to your perfect manual and it works great so far. But if I go downhill with a higher speed, 40km/h and above, scooter is shutting down when I break for a full stop. I set the ocd_current to 23A but the problem still exists. Im afraid of setting it higher because I dont want to toast my batteries. 
Im also not quite sure that the bms is shutting down, could also be the controller for all I know.
How can I brake with higher speeds and without the bms shutting down? Can you think of a solution?

Thank you

**Comment by BotoX** [comment]
Check the debug_info on the BMS, it has an error counter which should tell you what went wrong.
Charge overcurrent right now doesn't work, already implemented it though and will push it this weekend after testing.
And charge overcurrent also only works on C- anyways, your ESC should be connected on P-.
Check the readme about debug_print.

**Comment by videofunk** [comment]
Hello David, i drive a diffrent e-scooter it Austria known as freakyscooter. It uses the common brushless motor with chain and a very easy controler. I changed the acid battery into 15s6p 52volt 35A Lito. Normaly it was only 48v but they load up to 60 - 62volts so I went in that risk and it makes me 15% more speed.

But I still miss the quality of the smart xiaomi controler with BT and such... Do you ve an idea to bring both together



Am 13.01.2019 7:45 nachm. schrieb BotoX <notifications@github.com>:

Nope, not as it is right now anyways.
But it shouldn't be too hard to make that work, I don't have access to a BMS like that though.

**Comment by BotoX** [comment]
That is not something I'm interested in doing.
I'll probably focus on developing some things for my Bafang BBSHD ebike.

However you could change your motor controller to a better one, VESC is one example, check out forums like endless-sphere.com

**Comment by videofunk** [comment]
Yes, to change the components is the next. I thought to change to the  controler and FOC Motor, Minnimitors sells and use it with the frame. But anyways was good to meet the power of a higher rated Battery in that 48volts environment. I think higher constructed controllers limit the power to one series row like 10s or 14s but with the infantile oldschool version it is just limited to the shunt.  It hangs much more on  the powerconsumption. You feel it imidiatly if your Source gets weak on circles. With the new bms powered vehicles that us diffrent.

We made a bafang bike too, 48v 26"...but no way to drive faster then 40kmh. To dangerous!!! But same problems no BT no bms...
If you make a clue on that system, for sure a lot of people got very happy figure it out, me2.

I saw your counselor in TV after interview with trump. My god such a young Guy... Who votes in Austria?
Crazy.
________________________________
